pyuvdata/mwa_corr_fits.py

```diff
@@ -0,0 +1,482 @@
+            antenna_positions = np.zeros((len(antenna_numbers), 3))
+            antenna_positions[:, 0] = meta_tbl['East'][1::2]
+            antenna_positions[:, 1] = meta_tbl['North'][1::2]
+            antenna_positions[:, 2] = meta_tbl['Height'][1::2]
+
+            # TODO: self.antenna_diameters
+            # TODO: self.x_orientation
+
+        # reorder antenna parameters from metafits ordering
+        reordered_inds = antenna_numbers.argsort()
+        self.antenna_numbers = antenna_numbers[reordered_inds]
+        self.antenna_names = list(antenna_names[reordered_inds])
+        antenna_positions = antenna_positions[reordered_inds, :]
+        antenna_flags = antenna_flags[reordered_inds]
+        cable_lens = cable_lens[reordered_inds]
+
+        # find flagged antenna
+        flagged_ants = self.antenna_numbers[np.where(antenna_flags == 1)]
+
+        # set parameters from other parameters
+        self.Nants_data = len(self.antenna_numbers)
+        self.Nants_telescope = len(self.antenna_numbers)
+        self.Nbls = int(len(self.antenna_numbers) * (len(self.antenna_numbers) + 1) / 2.0)
+
+        # get telescope parameters
+        self.set_telescope_params()
+
+        # build time array of centers
+        time_array = np.arange(start_time + int_time / 2.0, end_time
+                               + int_time / 2.0 + int_time, int_time)
+
+        # convert from unix to julian times
+        julian_time_array = [Time(i, format='unix', scale='utc').jd
+                             for i in time_array]
+
+        # convert to integers
+        float_time_array = np.array([float(i) for i in julian_time_array])
+        # build into time array
+        self.time_array = np.repeat(float_time_array, self.Nbls)
+
+        self.Ntimes = len(time_array)
+
+        self.Nblts = int(self.Nbls * self.Ntimes)
+
+        # convert times to lst
+        self.lst_array = uvutils.get_lst_for_time(self.time_array,
+                                                  *self.telescope_location_lat_lon_alt_degrees)
+
+        # assumes no averaging
+        self.integration_time = np.array([int_time for i in range(self.Nblts)])
+
+        # convert antenna positions from enu to ecef
+        # antenna positions are "relative to
+        # the centre of the array in local topocentric \"east\", \"north\",
+        # \"height\". Units are meters."
+        antenna_positions_ecef = uvutils.ECEF_from_ENU(antenna_positions,
+                                                       *self.telescope_location_lat_lon_alt)
+        # convert to ITRF
+        self.antenna_positions = antenna_positions_ecef - self.telescope_location
+
+        # make initial antenna arrays, where ant_1 <= ant_2
+        ant_1_array = []
+        ant_2_array = []
+        for i in range(self.Nants_telescope):
+            for j in range(i, self.Nants_telescope):
+                ant_1_array.append(i)
+                ant_2_array.append(j)
```

Comment:
This looks suspicious to me, but maybe it's guaranteed to be correct because of ordering operations that are done later?